components/DetailCountry/index.js

```diff
@@ -37,7 +41,18 @@ export default function DetailCountry({ countries, deleteCountry }) {
   if (currentCountry === undefined) {
     return null;
   }
-  const { name, startDate, endDate, id, imagesUrls } = currentCountry;
+  function handleCurrentCountry() {
+    if (currentCountry) {
+      setIsCurrentCountry(currentCountry);
+    } else {
+      return null;
+    }
+  }
+
+  handleCurrentCountry();
```

Comment:
Dangerous bit of code. This could potentially result in an infinite loop.

You have a function which is immediately run and which sets a state variable. If you remember, _setting state triggers a re-render of your component._ Re-rendering runs the entire code again. The only reason why this is not an infinite loop is due to the state being set to the same value every time, which React ignores.

A better place would be the `useEffect` hook with an empty dependency array. But then again, you don't usually set state like this within useEffect either.

See if you can think through the app logic and figure out how to better manage your logic. Currently, it's not ideal and you might run into issues sometime down the line.